packages/web-components/@sfx/search-box/test/unit/search-box.test.ts

```diff
@@ -0,0 +1,193 @@
+        expect(searchbox.searchButton).to.equal(false);
+      });
+    });
+
+    describe('clearButton property', () => {
+      it('should have default value of false', () => {
+        expect(searchbox.clearButton).to.equal(false);
+      });
+    });
+  });
+
+  describe('connectCallback', () => {
+    it('should add an autocomplete_hover eventListener to the window', () => {
+      const windowAddEventListener = stub(window, 'addEventListener');
+
+      searchbox.connectedCallback();
+
+      expect(windowAddEventListener).to.have.been.calledWith(SEARCHBOX_EVENT.UPDATE_SEARCH_TERM, searchbox.updateText);
+    });
+  });
+
+  describe('disconnectCallback', () => {
+    it('should remove the autocomplete_hover eventListener from the window', () => {
+      const windowRemoveEventListener = stub(window, 'removeEventListener');
+
+      searchbox.disconnectedCallback();
+
+      expect(windowRemoveEventListener).to.have.been.calledWith(
+        SEARCHBOX_EVENT.UPDATE_SEARCH_TERM,
+        searchbox.updateText
+      );
+    });
+  });
+
+  describe('emitSearchEvent', () => {
+    it('should dispatch a search request event', () => {
+      const searchRequestEvent = new CustomEvent(SEARCHBOX_EVENT.SEARCH_REQUEST, {
+        detail: 'a',
+        bubbles: true
+      });
+      searchbox.value = 'a';
+
+      searchbox.emitSearchEvent();
+
+      expect(searchboxDispatchEvent).to.have.been.calledWith(searchRequestEvent);
+    });
+  });
+
+  describe('emitSearchBoxClearClick', () => {
+    it('should dispatch an emitSearchBoxClearClick', () => {
+      const searchboxClearedEvent = new CustomEvent(SEARCHBOX_EVENT.SEARCHBOX_CLEAR_CLICK);
+
+      searchbox.emitSearchBoxClearClick();
+
+      expect(searchboxDispatchEvent).to.have.been.calledWith(searchboxClearedEvent);
+    });
+  });
+
+  describe('updateText', () => {
+    it('should update the value property of the input element in response to data received', () => {
+      const detail = 'inputText';
+      const inputObject = { value: 'Placeholder text' };
+      const inputEvent = new CustomEvent(SEARCHBOX_EVENT.UPDATE_SEARCH_TERM, { detail: detail });
+      stub(searchbox, 'getInputElement').returns(inputObject);
+
+      searchbox.updateText(inputEvent);
+
+      expect(searchbox.value).to.equal(detail);
+      expect(searchbox.value).to.equal(inputObject.value);
+    });
+  });
+
+  describe('handleKeydown', () => {
+    it('should invoke the emitSearchEvent function if enter is pressed and value property length is greater than 0', () => {
+      const emitSearchStub = stub(searchbox, 'emitSearchEvent');
+      searchbox.value = 'hello';
+
+      searchbox.handleKeydown({ key: "Enter" });
+
+      expect(emitSearchStub).to.have.been.called;
+    });
+  });
+
+  describe('handleInput', () => {
+    it('should invoke the updateSearchTermValue function', () => {
+      const updateSearchTermValueStub = stub(searchbox, 'updateSearchTermValue');
+      const target = { value: 'dee' };
```

Comment:
`'dee'` should be factored out into a constant (and used in the assertion as well), but the `target` object can be inlined because it's not used more than once.